sources/Messages/Storages/Database/Sql.php

```diff
@@ -128,26 +130,84 @@ public function getAll(): Collection
-            `message`.`address_state_name`,
-            `message`.`phone`,
-            `message`.`email`,
-            `message`.`subject`,
-            `message`.`body`,
-            `message`.`date_hour`,
-            `message`.`status`
-        ';
+        $binds_array = array_keys($this->getBinds());
+        $fields = implode(", ", $binds_array);
+        $binds = implode(", ", $this->getBinds());
+
+        $statement = $this->pdo->prepare("
+            INSERT INTO {$this->table}
+            ({$fields})
+            VALUES ({$binds})
+        ");
+
+        $this->addBind($binds_array[0], PDO::PARAM_STR, $message->getName());
+        $this->addBind($binds_array[1], PDO::PARAM_STR, $message->getAddress()->getPlace());
+        $this->addBind($binds_array[2], PDO::PARAM_INT, $message->getAddress()->getNumber());
+        $this->addBind($binds_array[3], PDO::PARAM_STR, $message->getAddress()->getNeighborhood());
+        $this->addBind($binds_array[4], PDO::PARAM_STR, $message->getAddress()->getComplement());
+        $this->addBind($binds_array[5], PDO::PARAM_STR, $message->getAddress()->getCep());
+        $this->addBind($binds_array[6], PDO::PARAM_INT, $message->getAddress()->getCityId());
+        $this->addBind($binds_array[7], PDO::PARAM_STR, $message->getAddress()->getCityName());
+        $this->addBind($binds_array[8], PDO::PARAM_STR, $message->getAddress()->getStateName());
+        $this->addBind($binds_array[9], PDO::PARAM_STR, $message->getPhone());
+        $this->addBind($binds_array[10], PDO::PARAM_STR, $message->getEmail());
+        $this->addBind($binds_array[11], PDO::PARAM_STR, $message->getSubject());
+        $this->addBind($binds_array[12], PDO::PARAM_STR, $message->getBody());
+        $this->addBind($binds_array[13], PDO::PARAM_STR, $message->getDateHour()->format("Y-m-d H:i:s"));
+        $this->addBind($binds_array[14], PDO::PARAM_INT, $message->getStatus()->getValue());
+
+        $this->bind($statement);
+        
+        if ($statement->execute() === false) {
+            throw new Exception('ciebit.contactus.messages.storages.database.insert_error', 2);
+        }
+
+        return $this;
+    }
+
+    private function getBinds(): array
+    {
+        return [
+            'name' =>':name',
+            'address_place' =>':address_place',
+            'address_number' =>':address_number',
+            'address_neighborhood' =>':address_neighborhood',
+            'address_complement' =>':address_complement',
+            'address_cep' =>':address_cep',
+            'address_city_id' =>':address_city_id',
+            'address_city_name' =>':address_city_name',
+            'address_state_name' =>':address_state_name',
+            'phone' =>':phone',
+            'email' =>':email',
+            'subject' =>':subject',
+            'body' =>':body',
+            'date_hour' =>':date_hour',
+            'status' =>':status'
+        ];
+    }
+
+    private function getFields(string $alias=null): array
```

Comment:
Se ele irá retornar um array considero bem melhor ele retornar apenas o nome dos campos e o método que for usar adicionar o alias, ou como tinha proposto antes, ele retorna uma string que poderá ter ou não o *alias*. Como o nome dos campos será usado em outro lugar, podemos cogitar a possibilidade de criar um método que retorne apenas a lista de campos.

Reply:
Mas caso não seja passado nenhum parâmetro, é retornado apenas a lista de campos. O método que irá utilizá-lo pode passar ou não o parâmetro, caso queira um alias, @johnnysaymon

Reply:
Notei isso, mas o que achei estranho é ter métodos similares, já que irá retornar um *array* e necessitará concatenar para gerar o resultado desejado, o alias poderia ter sido adicionado nessa concatenação.

Reply:
Mas você acha que fica mais claro, para quem tá lendo o código pela primeira vez, a adição desse alias de qual modo? No modo atual, onde eu posso passar um parâmetro alias? Ou passando um string com o alias no método implode? @johnnysaymon

Reply:
Sem dúvidas da forma que está, por isso sugeri inicialmente que o retorno fosse uma string já pronta para inserção no SQL, assim como está o método de *where* ou *limit*.

Reply:
Então como irá ficar? Como está? Ou altero para retornar uma string @johnnysaymon

Reply:
Faz mais sentido para mim o retorno em formato de string.

Reply:
Rapaz, você havia dito em outro momento que o retorno em array era melhor, @johnnysaymon . Hahaha

Reply:
Acredito que causei confusão porque não estou restringindo a um único caminho, acabei sempre deixando aberto dois caminhos mas em ambos meu objetivo principal com isso tudo é evitar ter dois métodos contendo os mesmos dados, você poderá até manter os dois métodos e até com o mesmo retorno (embora não goste de alguns mas no momento não tenho outros argumentos para me colocar contra). O ponto é que a informação com o nome dos campos deverá ver de um único lugar, exemplos:

```php
public function getColumns(): array
{
    return [
        'id',
        'name',
        // ...
    ]
}

public function getFields(string $aliasTable = null): string
{
    $columns = $this->getColums();
    $alias = $aliasTable ? "`{$aliasTable}`." : '';
    return '`'. implode("`, {$alias}`", $columns) .'`';
}
```

Se quiser você ainda pode melhorar para o método *getBind* adicionando mais informações no método *getColumns*, como algo assim:

```php
public function getColumns(): array
{
    return [
        'id' => [PDO::PARAM_INT, 'getId'],
        'name' => [PDO::PARAM_STR, 'getName'],
        // ...
    ]
}
```